src/System.Private.CoreLib/shared/System/SpanHelpers.Char.cs

```diff
@@ -284,7 +284,7 @@ public static unsafe int IndexOf(ref char searchSpace, char value, int length)
                 if (offset < length)
                 {
                     Debug.Assert(length - offset >= Vector128<ushort>.Count);
-                    if ((((nint)Unsafe.AsPointer(ref searchSpace) + (nint)offset) & (nint)(Vector256<ushort>.Count - 1)) != 0)
+                    if (((nint)Unsafe.AsPointer(ref Unsafe.Add(ref searchSpace, (IntPtr)offset)) & (nint)(Vector256<ushort>.Count - 1)) != 0)
```

Comment:
This should read `Vector256<byte>.Count`, not `Vector256<ushort>.Count`. Reason is that alignment is measured in bytes, not number of elements.